Upgrade to Angular 16, Node.js 18, Bootstrap 5 and many other dependencies

swissbuechi opened the pull request:
Disclaimer: Goal is to reach Angular 18 or 19, but since we need to do step-by-step anyway, I figured Angular 16 is good progress coming from Angular 11.

## Updates

### Main updates

- Angular 11 to 16
- Node.js 14 to 18
- ng-bootstrap 9 to 15
  - Also needed bootstrap CSS update from 4 to 5 (from CDN to npm)
  - And bootswatch (from CDN to npm)
  - And a new dependency called popperjs
- Font awesome 5 to 6 (from CDN to npm)
- ynab 1 to 2
- d3 6 to 7

### Other updates (dev)
- eslint, jasmine, karma, etc... to latest version supported with Angular 16.

### Noticeable changes

- Bootstrap 5 spacings and font sizes vary a bit
- Bootswatch native dark mode for `flatly` instead of switching to the `darkly` theme.
- New navbar color that stays the same for light and dark mode to achieve consistent PWA user experience.
- Removed redundant `home` link navigation, just click on the logo or title.
- Fixed coloring of some buttons to streamline design.

JackMorrissey commented:
Wooooo. Good luck! When I merged your first PRs I took a stab at upgrading and in 30 minutes I barely made a dent to Angular 12 🤯

swissbuechi commented:
@JackMorrissey Haha thanks 😅

I've now got most of the styling issues fixed. Only the spacing is stil a bit messed up in some places.

Angular 16 should be ready to merge after putting in 1-2 more hours I guess 😄

swissbuechi commented:
@JackMorrissey Making good progress here. Only thing left to fix is the highlighting of the current navigation item in the nav bar and then Angular 16 + a ton of dependency updates would be ready to test and merge 😁 

I could deploy it on a github page for testing if you care?

swissbuechi commented:
@JackMorrissey I got a bit hysterical and accidentally added two small unnecessary changes to this PR which should've probably been in a dedicated PR...

- Removed the `home` from the navigation bar because users could just click the logo or title to get to the home section.

- Adjusted colors of the `Authenticate` button to match the design. (will post a screenshot after)

Do you want dedicated PRs or are we good?

JackMorrissey commented:
No worries. If this was work I'd say separate PRs, but this is Open Source. I'll take any free work you'd like to contribute haha.

swissbuechi commented:
Closed in favor of PR https://github.com/JackMorrissey/beyond-rule-4/pull/102 that upgrades all the way to Angular 19.
